### ISSUE TITLE
I can't receive notifications when app is open on android

### ISSUE THREAD

**Issue by beqramo** [customerio/customerio-reactnative#120]
Hi,

I integrated customer.io SDK and it has problems on Android. When the app is open it can't receive notifications in the foreground.
I followed  your recommendations here:
https://customer.io/docs/sdk/react-native/push/#cio-process-notification-payloads
but it doesn't work.
I think the problem is Intercom because it needs a native listener to have. (Also in parallel I'm using OneSignal too but it works well, I can receive notifications from Onesignal).
<img width="949" alt="image" src="https://user-images.githubusercontent.com/40486471/230648912-7b83cd5e-171f-47c3-a48d-b11d1be85483.png">
https://developers.intercom.com/installing-intercom/docs/react-native-push-notifications

On ios, it works well.

**Comment by mrehan27** [comment]
Hey @beqramo. Thank you for sharing the details. The customer.io docs you shared above are for processing push notifications from react native code files. If either Intercom or OneSignal provides callback to listen whenever a push notification is received, you can use Customer.io methods as explained in the [docs with example code](https://customer.io/docs/sdk/react-native/push/#cio-process-notification-payloads).

But since you are adding custom messaging service in native Android, I would recommend you to follow our docs in [Android SDK for receiving push notifications](https://www.customer.io/docs/sdk/android/push/#set-up-push) using custom service.

So, in your case, the service file should look something like:

```kotlin
class CustomFirebaseMessagingService : FirebaseMessagingService() {

 override fun onMessageReceived(message: RemoteMessage) {
    val didCustomerIOHandlePush = CustomerIOFirebaseMessagingService.onMessageReceived(context, message)
    if (didCustomerIOHandlePush) {
        // Push notification has been handled by Customer.io
    } else if (IntercomModule.isIntercomPush(remoteMessage)) {
        IntercomModule.handlePush(application, message)
    } else {
        // DO HOST LOGIC HERE
    }
 }
 
override fun onNewToken(token: String) {
    CustomerIOFirebaseMessagingService.onNewToken(context, token)
    IntercomModule.sendTokenToIntercom(application, refreshedToken)
}
```

Please let us know if this doesn't fix your issue.

**Comment by beqramo** [comment]
thanks will try but what should I import in the file?
this: 
`import io.customer.sdk.CustomerIO
`

**Comment by mrehan27** [comment]
@beqramo You should only need to import the following for Customer.io in above example
```
import io.customer.messagingpush.CustomerIOFirebaseMessagingService
```

Also, I just realized the issue you might be facing with imports is because we haven't yet considered exposing our native SDK dependencies to react native customers. We'll discuss this internally to consider exposing them in future. But for now, you'll have to add the `messaging-push-fcm` dependency to your app level `build.gradle` file (`android/app/build.gradle`) to make it accessible by your app code

```
implementation "io.customer.android:messaging-push-fcm"
```

For more details on adding native dependencies, please see [this comment](https://github.com/customerio/customerio-reactnative/issues/77#issuecomment-1409987498). Let us know if you have more questions.

**Comment by beqramo** [comment]
Hi again @mrehan27 
It doesn't work.
```
error: method onNewToken in class CustomerIOFirebaseMessagingService cannot be applied to given types;
    CustomerIOFirebaseMessagingService.onNewToken(getApplication(), refreshedToken);
```
I tried  this way too:
```
 error: non-static method onNewToken(String) cannot be referenced from a static context
    CustomerIOFirebaseMessagingService.onNewToken(refreshedToken);
```
this:
```


But it doesn't work
 CustomerIOFirebaseMessagingService.onNewToken(getContext(),refreshedToken);

```
package is loaded but calling to the functions are wrong.

**Comment by mrehan27** [comment]
Hey @beqramo. Sorry for the inconvenience caused due to this. I realized that we haven't added `JvmStatic` annotation to these methods which is restricting them to be used directly from Java classes 🤦🏻 We'll fix this in upcoming releases. For now, you can either create service in Kotlin or can call using `Companion` object from Java class as follows:

```
CustomerIOFirebaseMessagingService.Companion.onNewToken(getApplication(), "refreshedToken");
```

**Comment by mrehan27** [comment]
@beqramo Sorry for the typo earlier, I've updated that as well. You should pass the `refreshedToken` variable instead, so it should look like

```
CustomerIOFirebaseMessagingService.Companion.onNewToken(getApplication(), refreshedToken);
```

Rest seems good, it should be working, please let us know if this didn't work for you.

**Comment by SYoder1** [comment]
Hey @beqramo! Did you have to do anything to get Intercom and customerIO working on iOS? I can recive pushes from customerIO and the intercom ones are just lost

**Comment by Shahroz16** [comment]
@SYoder1 do you have another issue other than the [one](https://github.com/customerio/customerio-reactnative/issues/191) you created?

**Comment by SYoder1** [comment]
Yes, We have Intercom setup, and after setting up the CIO sdk, my Intercom notifications are not showing any more. I fixed it on Android before finding this thread (but solved it in a similar way), but ios they just disappear.

I tried to do something like this, but `didReceiveNotificationResponse` never gets called.
```
- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void(^)(void))completionHandler {
  NSDictionary *userInfo = response.notification.request.content.userInfo;
  if ([IntercomModule isIntercomPushNotification:userInfo]) {
    [IntercomModule handleIntercomPushNotification:userInfo];
    
    completionHandler();
  }
  else {
    [pnHandlerObj userNotificationCenter:center didReceiveNotificationResponse:response withCompletionHandler:completionHandler];
  }
}
```

**Comment by Shahroz16** [comment]
Can you please create a ticket for it? we can look into it.
